src/elements/content-sidebar/activity-feed/activity-feed/ActiveState.js

```diff
@@ -296,42 +300,66 @@ const ActiveState = ({
                                 isFocused={isFocused}
                                 ref={refValue}
                             >
-                                <ActivityThread
-                                    data-testid="activity-thread"
-                                    currentUser={currentUser}
-                                    getAvatarUrl={getAvatarUrl}
-                                    getMentionWithQuery={getMentionWithQuery}
-                                    getUserProfileUrl={getUserProfileUrl}
-                                    hasNewThreadedReplies={hasNewThreadedReplies}
-                                    hasReplies={hasReplies}
-                                    isPending={item.isPending}
-                                    isRepliesLoading={item.isRepliesLoading}
-                                    mentionSelectorContacts={mentionSelectorContacts}
-                                    onHideReplies={onHideRepliesHandler(item.id)}
-                                    onReplyCreate={onReplyCreateHandler(item.id, item.type)}
-                                    onReplyDelete={onReplyDeleteHandler(item.id)}
-                                    onReplyEdit={onReplyUpdateHandler(item.id)}
-                                    onReplySelect={onCommentSelectHandler(item.id)}
-                                    onShowReplies={onShowRepliesHandler(item.id, item.type)}
-                                    repliesTotalCount={item.total_reply_count}
-                                    replies={item.replies}
-                                    translations={translations}
-                                >
-                                    <AnnotationActivity
+                                {hasNewThreadedReplies && onAnnotationSelect ? (
+                                    // TODO: legitimate, pre-existing typing issue that was previously undetected
+                                    // Conflict between BoxCommentPermissions and BoxTaskPermissions
+                                    // $FlowFixMe
+                                    <BaseComment
+                                        {...commentAndAnnotationCommonProps}
+                                        annotationActivityLink={
+                                            <AnnotationActivityLinkProvider
+                                                item={item}
+                                                onSelect={onAnnotationSelect}
+                                                isCurrentVersion={currentFileVersionId === itemFileVersionId}
+                                            />
+                                        }
+                                        onAnnotationEdit={onAnnotationEdit}
```

Comment:
Do we need to specify an onAnnotationEdit and an onCommentEdit?  Can we just have one onEdit function and assign the proper handler to it?

Reply:
Well, `onAnnotationEdit` and `onCommentEdit` edit take different args, so I'm resolving that within BaseComment.js. We could potentially do it in ActiveState (or in my previous PR I handled it in `BaseCommentWrapper`). Good eye though. The reason that both props are passed here was because I was trying out passing `onCommentEdit` down to `Replies` (within `BaseComment`) to see if that would enable modifying replies. It didn't. I thought I cleaned everything up, but `onCommentEdit` doesn't need to be defined here.

Reply:
I removed `onCommentEdit` here.